allensdk/brain_observatory/behavior/session_apis/data_io/behavior_ophys_nwb_api.py

```diff
@@ -311,3 +327,171 @@ def get_motion_correction(self) -> pd.DataFrame:
         motion_correction_data['y'] = ophys_module.get_data_interface('ophys_motion_correction_y').data[:]
 
         return pd.DataFrame(motion_correction_data)
+
+    def get_rig_metadata(self) -> Optional[dict]:
+        try:
+            et_mod = self.nwbfile.get_processing_module("eye_tracking_rig_metadata")
+        except KeyError as e:
+            warnings.warn(f"This ophys session '{int(self.nwbfile.identifier)}' has no eye tracking rig metadata. "
+                          f"(NWB error: {e})")
+            return None
+
+        meta = et_mod.get_data_interface("eye_tracking_rig_metadata")
+
+        rig_geometry = pd.DataFrame({
+            f"monitor_position_{meta.monitor_position__unit_of_measurement}": meta.monitor_position,
+            f"camera_position_{meta.camera_position__unit_of_measurement}": meta.camera_position,
+            f"led_position_{meta.led_position__unit_of_measurement}": meta.led_position,
+            f"monitor_rotation_{meta.monitor_rotation__unit_of_measurement}": meta.monitor_rotation,
+            f"camera_rotation_{meta.camera_rotation__unit_of_measurement}": meta.camera_rotation
+        })
+
+        rig_geometry = rig_geometry.rename(index={0: 'x', 1: 'y', 2: 'z'})
+
+        returned_metadata = {
+            "geometry": rig_geometry,
+            "equipment": meta.equipment
+        }
+
+        return returned_metadata
+
+    def get_screen_gaze_data(self, include_filtered_data=False) -> Optional[pd.DataFrame]:
```

Comment:
A lot of the contents of `get_screen_gaze_data` and `get_pupil_data` are almost exactly the same as the ecephys implementation (I think?). Perhaps it might be worthwhile to create a MixIn class that could be re-used between behavior+ophys and ecephys?

If that sounds like too much effort for now, we can maybe make a ticket and perhaps address in the future?

Reply:
Let's shelve this issue for another day.

Reply:
I agree, but I don't know if I have enough knowledge of how ecephys and ophys relate to know how to organize the classes in order to share logic